sdks/python/apache_beam/ml/anomaly/transforms.py

```diff
@@ -600,20 +610,43 @@ def expand(
     #
     # We select uuid.uuid1() for its inclusion of node information, making it
     # more suitable for parallel execution environments.
-    add_temp_key_fn: Callable[[InputT], KeyedInputT] \
-        = lambda e: (e[0], (str(uuid.uuid1()), e[1]))
-    keyed_input = (input | "Add temp key" >> beam.Map(add_temp_key_fn))
 
+    if isinstance(input.element_type, TupleConstraint):
+      keyed_input = input
+    else:
+      # Add a None key if the input is unkeyed.
+      keyed_input = input | beam.WithKeys(None)
```

Comment:
Nit: I'd shy away from a `None` key and use a real-valued constant. Once again not wholly necessary, just a thought